### PR TITLE
docs: general formatting and grammar improvements

### PR DESCRIPTION
**Pull request by SethFalco**
**What is this feature?**

Sorry for the really trivial PR!

I was just reading through the docs for Grafana. Just noticed some things that made reading it a bit uncomfortable, so I was fixing them along the way.

Some of the changes are subjective, so feel free to ask me to undo any particular change.

Most formatting changes were the result of my editor applying your `.editorconfig`. (i.e. trimming whitespace, or adding a new line at the end of a file.)

### Incorrect capitalization of nouns

* Github → GitHub
* docker → Docker
* Hashicorp → HashiCorp

### Oxford comma

Weather a serial comma is used or not is fairly inconsistent. I opted to add them where they were missing in some places.

### Link docker group

> add your user to the `docker` group

Links this to the first-party documentation explaining what it is and caveats around it.
It's mostly, so users can see this warning:

> The docker group grants privileges equivalent to the root user. For details on how this impacts security in your system, see [Docker Daemon Attack Surface](https://docs.docker.com/engine/security/#docker-daemon-attack-surface).
> 
> — https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user

Nothing wrong with that, but just to make sure users are aware before they blindly add users to the docker group.

### Misc

And various other changes:
* Flat out typos (`file that you are modify by removing`)
* Uppercasing acronyms
* Using contractions where they are more comfortable to read
* Replacing words/terms with alternatives that are more common or familiar for the given context.
* Avoid starting sentences with a lowercase character.
* Very minor change in CSS which just swapped the single/double quotes.

**Why do we need this feature?**

Nice to have.

**Who is this feature for?**

People reading docs.

**Which issue(s) does this PR fix?**:

N/A

Fixes #

**Special notes for your reviewer**:

This is basically just a reopening of the following but rebased:
* https://github.com/grafana/grafana/pull/52967

Some changes were scrapped because either the documentation has just changed since then, or I deemed them too subjective.

### CODE REVIEW

**Review by chri2547 — COMMENTED** [review]
Very nice changes! Thank you so much for the contribution. I'm going to have engr sign off on these changes and then we'll merge in. That will likely happen next week. Thanks again!

**Comment by SethFalco** [comment]
Hey hey! Sounds good, just wanted to make a small note as well.

If possible, I've love if these could be reviewed in future too. I did them a while back but just had to rebase as there were lots of changes. ^-^'

* https://github.com/grafana/github-datasource/pull/177
* https://github.com/grafana/github-datasource/pull/178

**Review by ryantxu — APPROVED** [review]
LGTM

**Review by nmarrs — APPROVED** [review]
LGTM

**Review by zerok — APPROVED** [review]

**Comment by zerok** [comment]
Thank you 😍

File: pkg/login/social/github_oauth.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func (s *SocialGithub) UserInfo(client *http.Client, token *oauth2.Token) (*Basi
 
 	role, grafanaAdmin := s.extractRoleAndAdmin(response.Body, teams, true)
 	if s.roleAttributeStrict && !role.IsValid() {
-		return nil, &InvalidBasicRoleError{idP: "Github", assignedRole: string(role)}
+		return nil, &InvalidBasicRoleError{idP: "GitHub", assignedRole: string(role)}
```
Comment (joshhunt) [inline pkg/login/social/github_oauth.go:200]: @grafana/grafana-authnz-team Are you able to review this change? It looks like this property is only used for an error message (and it's a meaningful string/'enum'), but would like to be extra sure :)

**Review by joshhunt — COMMENTED** [review]
👍🏻  Reviewed everything except for .md files - All code changes looks fine for me. Request @grafana/grafana-authnz-team to double check the change in github_oauth.go :)

File: pkg/login/social/github_oauth.go (hunk 1 above)
Comment (SethFalco) [reply pkg/login/social/github_oauth.go:200]: Ahh, sorry about that.
Just let me know if it needs to be undone.

With bulk changes, I usually only target docs/html/comments/prints/etc, so this one was a mistake.

Comment (kalleep) [reply pkg/login/social/github_oauth.go:200]: Yes this is only used for error message so the change is fine :)

**Review by chri2547 — APPROVED** [review]

**Comment by jdbaldry** [comment]
> Token is empty, can't dispatch event. This is expected for PRs coming from forks, please check that the changes are compatible with Enterprise before merging this PR.
> /home/runner/work/grafana/grafana/actions/node_modules/@actions/core/lib/core.js:106
>        throw new Error(`Input required and not supplied: ${name}`);

@grafana/grafana-release-eng Would somebody be able to advise on how we should validate this PR so it can be merged?

**Comment by chri2547** [comment]
@jdbaldry , Grafana Enterprise check seems to be stuck.

**Pull request closed without merging**